skipruntime-ts/ts/src/internals/skipruntime_types.ts

```diff
@@ -112,6 +112,11 @@ export interface Context {
     rangeOpt?: [K, K][] | null,
   ) => void;
 
+  union: <K extends TJSON, V extends TJSON>(
+    name: string,
+    ...collections: EagerCollection<K, V>[]
```

Comment:
I don't have a strong opinion on this, I'm still not sure of why we have so many layers, but it looks like at this level, we're already in the world of strings and not collections anymore (the `import type EagerCollection` triggered me)

Reply:
What is the name for? I thought those where generated for the user in the TS layer.

Reply:
This is from a previous version -- the current diff generates the name on the skip side.